### PR TITLE
Role based titles for Breadcrumbs and Tombstone

### PR DESCRIPTION
**Pull request by dimak1**
*Issue #:* /bcgov/entity#11644
*Issue #:* /bcgov/entity#11645

*Description of changes:*

- Add a `userAccessRole` to define which combo/single access role user has (MHR, PPR, MHR+PPR, etc.)
- Dynamically display titles for Breadcrumbs and Tombstone based on the `userAccessRole`
- Define Breadcrumbs and Tombstone titles as resources
- Update tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/store/actions/actions-model.ts [hunk 1]
```diff
@@ -56,6 +56,17 @@ export const setOriginalAddCollateral: ActionIF = ({ commit }, addCollateral: Ad
...
+export const setUserAccessRole: ActionIF = ({ commit }, authRoles: Array<string>): void => {
+  const accessRole = []
+
+  if (authRoles.includes('staff')) accessRole.push('STAFF')
+  else accessRole.push('CLIENT')
+
+  if (authRoles.includes('ppr')) accessRole.push('PPR')
+  if (authRoles.includes('mhr')) accessRole.push('MHR')
+  commit('mutateUserAccessRole', accessRole.join('_'))
+}
```
Comment (cameron-eyds) [inline ppr-ui/src/store/actions/actions-model.ts:68]: Just curious, how does this role array differs from the roles in `authRoles`. 
Do the store getters between lines between 213 - 243 do the same thing?

File: ppr-ui/src/components/tombstone/TombstoneDefault.vue [hunk 2]
```diff
@@ -73,47 +74,23 @@ export default defineComponent({
...
-        } else if (isRoleStaffPpr.value || isRoleStaff.value) {
-          header = 'Staff Personal Property Registry'
-        } else if (isRoleClientMhrPpr.value) {
-          header = 'My Manufactured Home and Personal Property Registry'
-        } else if (isRoleClientMhr.value) {
-          header = 'My Manufactured Home Registry'
-        } else if (isRoleClientPpr.value) {
-          header = 'My Personal Property Registry'
-        }
-        return header
```
Comment (cameron-eyds) [inline ppr-ui/src/components/tombstone/TombstoneDefault.vue:116]: Are these getters still being used elsewhere?

File: ppr-ui/src/resources/breadcrumbsTitles.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export const breadcrumbsTitles = {
+  CLIENT_PPR: 'My Personal Property Registry',
+  CLIENT_MHR: 'My Manufactured Home Registry',
+  CLIENT_PPR_MHR: 'My Asset Registries',
+  STAFF_PPR: 'Staff Personal Property Registry',
+  STAFF_MHR: 'Staff Manufactured Home Registry',
+  STAFF_PPR_MHR: 'Staff Asset Registries'
+}
```
Comment (cameron-eyds) [inline ppr-ui/src/resources/breadcrumbsTitles.ts:8]: I like the approach but why as resource consts vs enums?
Same question for `tombstoneTiles`

**Review by cameron-eyds — COMMENTED** [review]
Looks good so far, just had a few questions/comments.

**Review by doug-lovett — APPROVED** [review]
No comments in addition to Cameron's.

File: ppr-ui/src/store/actions/actions-model.ts (hunk 1 above)
Comment (dimak1) [reply ppr-ui/src/store/actions/actions-model.ts:68]: `authRoles` has each role separately e.g. ["staff","ppr","mhr"]. So I wanted to create a one user access combo role e.g. 'STAFF_PPR_MHR' to use it as a _key_ to get content _values_ from resources.

File: ppr-ui/src/components/tombstone/TombstoneDefault.vue (hunk 2 above)
Comment (dimak1) [reply ppr-ui/src/components/tombstone/TombstoneDefault.vue:116]: They shouldn't be used, I created them recently and now removed them after refactoring.

File: ppr-ui/src/resources/breadcrumbsTitles.ts (hunk 3 above)
Comment (dimak1) [reply ppr-ui/src/resources/breadcrumbsTitles.ts:8]: I felt like these are client-facing content titles should be part of the resources, but I'm open to discussion.

**Comment by dimak1** [comment]
@cameron-freshworks addressed your comments.

File: ppr-ui/src/components/tombstone/TombstoneDefault.vue (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/tombstone/TombstoneDefault.vue:116]: Sorry, i was referring to the global getters created for these conditionals
ie `isRoleStaffMhrPpr` `isRoleClientMhrPpr` etc
Just seen that you've taken a different approach, so do we still need those global getters?

File: ppr-ui/src/resources/breadcrumbsTitles.ts (hunk 3 above)
Comment (cameron-eyds) [reply ppr-ui/src/resources/breadcrumbsTitles.ts:8]: That makes sense 👍

File: ppr-ui/src/store/actions/actions-model.ts (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/store/actions/actions-model.ts:68]: Sounds good, i like it. 
Same comment as above, with this approach we might be able to remove those previously added global getters, unless they are used elsewhere 👍

File: ppr-ui/src/components/tombstone/TombstoneDefault.vue (hunk 2 above)
Comment (dimak1) [reply ppr-ui/src/components/tombstone/TombstoneDefault.vue:116]: Good catch, I didn't remove them from there, thanks!

Comment (cameron-eyds) [reply ppr-ui/src/components/tombstone/TombstoneDefault.vue:116]: right on, resolved!

**Review by cameron-eyds — APPROVED** [review]
Looks good 👍

**Comment by dimak1** [comment]
@cameron-freshworks I've removed global setters and introduced a 'data-test-id' attr for testing.

**Comment by cameron-eyds** [comment]
> @cameron-freshworks I've removed global setters and introduced a 'data-test-id' attr for testing.

Sounds good! 👍 
Thanks

**Pull request merged**